fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/BrowserToolbarMenuController.kt

```diff
@@ -254,6 +256,18 @@ class DefaultBrowserToolbarMenuController(
                     )
                 }
             }
+            is ToolbarMenu.Item.OpenInRegularTab -> {
+                // Even if we could "move" the tab (ie, allow it to keep the same tab ID and some
+                // session data) we'd choose to not do that - the possibility that other existing
+                // private tabs have this tab ID as the parent is one example of some of the risks
+                // we'd need to mitigate. So instead we dispatch "remove" and "add" actions.
+                currentSession?.let { session ->
+                    getProperUrl(session)?.let { url ->
+                        store.dispatch(TabListAction.RemoveTabAction(session.id, false))
+                        store.dispatch(TabListAction.AddTabAction(createTab(url), true))
+                    }
+                }
```

Comment:
Nice work on figuring out a solution for those edge cases! Can we move this to the `TabsUseCases` so that it can be reusable code if we need to move this around or use it in multiple places?

Reply:
A UseCase would probably be valid here for a few reasons, I think: we have UseCases that only one line of code to make sharing of repeatable code easier amongst components, we can separate the testable parts of the code (these two actions) from the UI, and this is probably not the final destination of this change.

> It looks like the use-case would require as params the store, the tab ID and the tab URL (the latter because getProperUrl() is private) - while that would work, it seems like an odd signature (you'd expect the ID would be enough to give you the URL, but as above, getProperSession() can't be called by the use-case).

Yeah, that's fair! The `getProperSession` is an artifact from our reader view web extension integration so making that a nullable parameter would work for today and then when the new ReaderView GV implementation lands, it will take out that weird method and this param (we can file a bug with a dependant on BZ-1770985).

I was curious and took an opportunity to write out what this use case would look like with a few extra checks. You don't have to use this, but I think it was easier to write it out as code instead of words. 😄 

```kotlin
    class MigratePrivateTabUseCase(
        private val store: BrowserStore,
    ) {
        /**
         * @param alternativeUrl url to load if one from the tab does not exist.
         * @return the ID of the tab that was re-created as part of the move.
         */
        operator fun invoke(
            tabId: String,
            alternativeUrl: String? = null
        ): String {
            val tab = store.state.findTab(tabId) ?: throw IllegalStateException("Tab does not exist.")

            if (!tab.content.private) {
                throw IllegalArgumentException("The tab we are trying to move is not private.")
            }

            val url = alternativeUrl ?: tab.content.url
            val newTab = createTab(url)

            store.dispatch(TabListAction.RemoveTabAction(tabId, false))
            store.dispatch(TabListAction.AddTabAction(newTab, true))

            return newTab.id
        }
    }
```